### DISCUSSION TITLE
Automatically broadcast MIDI service using ESP8266

### DISCUSSION THREAD

**Discussion by svanimisetti** [lathoub/Arduino-AppleMIDI-Library#121] [Q&A] [answered]
I want ESP8266 to broadcast its service on the network. When I start [MIDI network](https://apps.apple.com/us/app/midi-network/id1102428729) app - it automatically lists the rtpMIDI session from my desktop (Tobias' app). However, for ESP8266 I have to manually enter the address and port. Is there a way I can make ESP8266 "broadcast" its service so that I don't have to enter IP/port manually. The reason being that when I connect to my wifi network - the router assigns a different IP to ESP8266 everytime - and with the current example - I have to obtain the IP address from text printed to serial port. I want to make my MIDI device standalone - and not have to use a PC attached to it.  Is this even possible?

**Comment by lathoub** [comment, marked as answer]
Broadcasting the RTP-MIDI service is a mDNS function, and not a core part of the RTP-MIDI spec (albeit very handy and useful).

See the [mDNS example for SAMD](https://github.com/lathoub/Arduino-AppleMIDI-Library/blob/master/examples/SAMD_Bonjour/SAMD_Bonjour.ino) in this lib. ESP8266 mDNS example (and lib) [here](https://github.com/esp8266/Arduino/blob/master/libraries/ESP8266mDNS/examples/mDNS_Web_Server/mDNS_Web_Server.ino
)

**Reply by svanimisetti** [reply]
Thanks for clarifying. I assumed that since apple-midi is already a part of the [Bonjour service list](http://www.dns-sd.org/ServiceTypes.html), it would be automatically detected if any node on my network is running the Bonjour service. For example, I have it running on my Win10 desktop running iTunes+Bonjour, which automatically detected my iPhone whenever I connect it to my network. I will try to cook up an example and see if it works. If I get it working, I will add both the SoftAP (#117) and auto-discovery/broadcast examples to your repo.

**Reply by lathoub** [reply]
> it would be automatically detected

Bonjour works the other way around: devices announce themselves (they are not _discovered_).

I look forward to your PR for ESP8266 SoftAP with mDNS

**Reply by svanimisetti** [reply]
@lathoub - just sent a PR creating a new example file for ESP8266. Tested on iOS (iPad/iPhone) and Win10 (rtpMIDI - although may need to relook this).
